config/import-control.xml

```diff
@@ -7,6 +7,8 @@
     <allow pkg="java.io"/>
     <allow pkg="java.util"/>
     <allow pkg="com.github.checkstyle.regression.data"/>
+    <allow pkg="org.immutables"/>
```

Comment:
> should only be specifically allowed for the class 

This isn't allowed yet as it is still an open issue at https://github.com/checkstyle/checkstyle/issues/3492 .
Move this to `data` package and make a comment above line that it is only allowed until `https://github.com/checkstyle/checkstyle/issues/3492`, if there is no other way to signify it as a `TypeAdapters`.